File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    let match = 0;
    let maybe = 0;
    for (let num of lottos) {
        if (num == 0) {
            maybe ++
        } else if (win_nums.indexOf(num) != -1) {
            match ++
        }
    }
    const rank = (cnt) => cnt > 1 ? 7 - cnt : 6
    return [rank(match + maybe), rank(match)];
}

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    correct = 0
    maybe = 0
    for num in lottos:
        if num in win_nums:
            correct += 1
            maybe += 1
        if num == 0:
            maybe += 1
    if maybe > 1:
        answer.append(7-maybe)
    else:
        answer.append(6)
    if correct > 1:
        answer.append(7-correct)
    else:
        answer.append(6)
    return answer